lib/v2/penguin-random-house/utils.js

```diff
@@ -0,0 +1,98 @@
+const cheerio = require('cheerio');
+const got = require('@/utils/got');
+const { art } = require('@/utils/render');
+const path = require('path');
+const dateParser = require('@/utils/dateParser');
+
+const parseBookInList = (element) => {
+    const $ = cheerio.load(element);
+    const title = $('h2').first().text();
+    const author = $('h2.author').first().text();
+    const description = $('.desc').first().text();
+    let imageSrc = $('img.cover__backcover').attr('src');
+    let imageAlt = $('img.cover__backcover').attr('alt');
+    if (!imageSrc) {
+        imageSrc = $('img.img-responsive').attr('data-src');
+        imageAlt = $('img.img-responsive').attr('alt');
+    }
+
+    return art(path.join(__dirname, 'templates/book.art'), {
+        title,
+        author,
+        description,
+        imageSrc,
+        imageAlt,
+    });
+};
+
+const parseBooks = (element) => {
+    const $ = cheerio.load(element);
+    const description = $('h2.read-down-text').first().html();
+    let mainBlock = '';
+
+    $('.awesome-list>li').map((i, element) => {
+        const appending = parseBookInList(element);
+        mainBlock += appending;
+        return appending;
+    });
+
+    return {
+        description,
+        content: mainBlock,
+    };
+};
+
+const parseArticle = (element) => {
+    const $ = cheerio.load(element);
+    const description = $('h2.hdr-smalltxt').first().html();
+    const imageSrc = $('div.img-block>img').first().attr('src');
+    const imageAlt = $('div.img-block>img').first().attr('alt');
+    let mainBlock = '';
+
+    const descriptionBlock = art(path.join(__dirname, 'templates/book.art'), {
+        description,
+        imageSrc,
+        imageAlt,
+    });
+
+    $('div.main-content>p,div.main-content>ul').map((i, element) => {
+        const appending = cheerio.load(element);
+        mainBlock += appending.html();
+        return appending;
+    });
+
+    return {
+        description: descriptionBlock,
+        content: mainBlock,
+    };
+};
+
+const parseList = async (items, ctx, contentParser) =>
+    await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.url, async () => {
+                const itemRes = await got({
+                    method: 'get',
+                    url: item.url,
+                });
+
+                const itemPage = itemRes.data;
+                const itemParsed = contentParser(itemPage);
+
+                const result = {
+                    title: item.title,
+                    description: itemParsed.description + '<br>' + itemParsed.content,
+                    pubDate: dateParser(new Date().toISOString()), // No time
```

Comment:
1. https://github.com/DIYgod/RSSHub/pull/10287#discussion_r928079587
2. Don't add a pubDate if there isn't a date.
https://docs.rsshub.app/joinus/pub-date.html

The post date is available from 
![image](https://user-images.githubusercontent.com/11386903/180604946-8816adb9-8e1f-438c-a4dc-c5ab1b0565e7.png)